flink-table-store-core/src/main/java/org/apache/flink/table/store/table/system/FilesTable.java

```diff
@@ -0,0 +1,335 @@
+
+        private FilesRead(SchemaManager schemaManager) {
+            this.schemaManager = schemaManager;
+        }
+
+        @Override
+        public TableRead withFilter(Predicate predicate) {
+            // TODO
+            return this;
+        }
+
+        @Override
+        public TableRead withProjection(int[][] projection) {
+            this.projection = projection;
+            return this;
+        }
+
+        @Override
+        public RecordReader<InternalRow> createReader(Split split) throws IOException {
+            if (!(split instanceof FilesSplit)) {
+                throw new IllegalArgumentException("Unsupported split: " + split.getClass());
+            }
+            FilesSplit filesSplit = (FilesSplit) split;
+            DataTableScan.DataFilePlan dataFilePlan = filesSplit.dataFilePlan();
+            List<Iterator<InternalRow>> iteratorList = new ArrayList<>();
+            for (DataSplit dataSplit : dataFilePlan.splits) {
+                iteratorList.add(
+                        Iterators.transform(
+                                dataSplit.files().iterator(),
+                                v ->
+                                        toRow(
+                                                dataSplit,
+                                                dataFilePlan.snapshotId,
+                                                v,
+                                                filesSplit.storeTable)));
+            }
+            Iterator<InternalRow> rows = Iterators.concat(iteratorList.iterator());
+            if (projection != null) {
+                rows =
+                        Iterators.transform(
+                                rows, row -> ProjectedRow.from(projection).replaceRow(row));
+            }
+            return new IteratorRecordReader<>(rows);
+        }
+
+        private InternalRow toRow(
+                DataSplit dataSplit,
+                Long snapshotId,
+                DataFileMeta dataFileMeta,
+                FileStoreTable storeTable) {
+            TableSchema tableSchema =
+                    schemaManager.schema(
+                            storeTable.snapshotManager().snapshot(snapshotId).schemaId());
+            RowDataToObjectArrayConverter partitionConverter =
+                    new RowDataToObjectArrayConverter(tableSchema.logicalPartitionType());
+
+            TableSchema dataSchema = schemaManager.schema(dataFileMeta.schemaId());
+            RowType keysType = dataSchema.logicalTrimmedPrimaryKeysType();
+            RowDataToObjectArrayConverter keyConverter =
+                    keysType.getFieldCount() > 0
+                            ? new RowDataToObjectArrayConverter(
+                                    dataSchema.logicalTrimmedPrimaryKeysType())
+                            : new RowDataToObjectArrayConverter(dataSchema.logicalRowType());
+
+            int[] indexMapping =
+                    SchemaEvolutionUtil.createIndexMapping(
```

Comment:
I don't want to expose `SchemaEvolutionUtil` things here.
Can we extract a some class like `FieldStatsConverters` to do this work?
```
class FieldStatsConverters {
   FieldStatsConverters(Function<Long, TableSchema> schemas, long tableSchemaId) {...}
   FieldStatsArraySerializer get0rCreate(long fileSchemaId);
}
```